docs/src/content/docs/guides/pages.mdx

```diff
@@ -0,0 +1,137 @@
+
+<FileTree>
+
+- src/
+  - content/
+    - docs/
+      - hello-world.md
+      - reference/
+        - faq.md
+
+</FileTree>
+
+### Type-safe frontmatter
+
+All Starlight pages share a customizable [common set of frontmatter properties](/reference/frontmatter/) to control how the page appears:
+
+```md
+---
+title: Hello, World!
+description: This is a page in my Starlight-powered site
+---
+```
+
+If you forget anything important, Starlight will let you know.
+
+## Custom pages
+
+For advanced use cases, you can create custom pages in Astro’s `src/pages/` directory.
+This is helpful if you need to build pages with a completely custom layout or generate a page from an alternative data source.
+The `src/pages/` directory uses file-based routing and includes support for `.astro` files.
+
+Read more in the [“Pages” guide in the Astro docs](https://docs.astro.build/en/basics/astro-pages/).
+
+### Using Starlight’s design in custom pages
+
+To use the Starlight layout in custom pages, wrap your page content with the `<StarlightPage />` component.
+This can be helpful if you are generating content dynamically but still want to use Starlight’s design.
+
+```astro
+---
+// src/pages/custom-page/example.astro
+import StarlightPage from '@astrojs/starlight/components/StarlightPage.astro';
+import CustomComponent from './CustomComponent.astro';
+---
+
+<StarlightPage frontmatter={{ title: 'My custom page' }}>
+	<p>This is a custom page with a custom component:</p>
+	<CustomComponent />
+</StarlightPage>
+```
+
+#### Props
+
+The `<StarlightPage />` component accepts the following props.
+
+##### `frontmatter` (required)
+
+**type:** `StarlightPageFrontmatter`
+
+Set the [frontmatter properties](/reference/frontmatter/) for this page, similar to frontmatter in Markdown pages.
+The [`title`](/reference/frontmatter/#title-required) property is required and all other properties are optional.
+
+The following properties differ from Markdown frontmatter:
+
+- The [`slug`](/reference/frontmatter/#slug) property is not supported and is automatically set based on the custom page’s URL.
+- The [`editUrl`](/reference/frontmatter/#editurl) option requires a URL to display an edit link.
+- The [`sidebar`](/reference/frontmatter/#sidebar) property is not supported. In Markdown frontmatter, this option allows customization of [autogenerated link groups](/reference/configuration/#sidebar), which is not applicable to pages using the `<StarlightPage />` component.
+
+##### `sidebar`
+
+**type:** `SidebarEntry[] | undefined`  
+**default:** the sidebar generated based on the [global `sidebar` config](/reference/configuration/#sidebar)
+
+Provide a custom site navigation sidebar for this page.
+If not set, the page will use the default global sidebar.
+
+##### `hasSidebar`
+
+**type:** `boolean`  
+**default:** `false` if [`frontmatter.template`](/reference/frontmatter/#template) is `'splash'`, otherwise `true`
+
+Set if the sidebar should be displayed on this page.
```

Comment:
```suggestion
Whether or not the sidebar should be displayed on this page.
```

Similarly, as a more concrete example, this would be another way to frame all these options. Not a deal breaker, just pointing out that these rely on the type and default to do all the defining here, vs explainy text. Simply observing that it stood out to me!

Reply:
```suggestion
Control whether or not the sidebar should be displayed on this page.
```